### PR TITLE
ddl: make rename a not exists table report right error message

### LINKED ISSUE #29893: Rename a non exist table report wrong error message
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
create database db;
use db;
rename table tb1 to tb2;
```

### 2. What did you expect to see? (Required)
```
MySQL 5.7
ERROR 1017 (HY000): Can't find file: './db_optimistic/tb1.frm' (errno: 2 - No such file or directory)
```
```
MySQL 8.0
ERROR 1146 (42S02): Table 'db_optimistic.tb1' doesn't exist
```
### 3. What did you see instead (Required)
```
TiDB 4.0.11
ERROR 1017 (HY000): Can't find file: 'db_optimistic' (errno: {%!d(string=tb1) %!d(string=tb1)} - %!s(MISSING))
```

### PR DESCRIPTION
**Pull request by fanrenhoo**
### What problem does this PR solve?

Issue Number: close #29893

Problem Summary: rename non exist table wrong error message

### What is changed and how it works?
1. in ddl/ddl_api.go, line:5236, I changed the err message as a "table not exist"
    I think the mysql 8.0 message is better than mysql 5.7. At first, I did change the msg in errno/errname.go as mysql5.7 (as "mysql.Message("Can't find file: './%-.192s/%-.192s.frm' (errno....."), but later I use "table not exist" because that clear and compatible for future mysql 8.0 directly.
   And i did not remove the always false line "isAlterTable :=false", because that way I may need to transfer hardcode "false" in func call, so I just leave it there as before  
  
2. in ddl/db_test.go, I add a test also. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
make rename table report a reasonable error message
```

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -5233,7 +5233,7 @@ func extractTblInfos(is infoschema.InfoSchema, oldIdent, newIdent ast.Ident, isA
 		if tableExists(is, newIdent, tables) {
 			return nil, 0, infoschema.ErrTableExists.GenWithStackByArgs(newIdent)
 		}
-		return nil, 0, errFileNotFound.GenWithStackByArgs(oldIdent.Schema, oldIdent.Name)
+		return nil, 0, infoschema.ErrTableNotExists.GenWithStackByArgs(oldIdent.Schema, oldIdent.Name)
```
Comment (xiongjiwei) [inline ddl/ddl_api.go:5236]: there is another usage at L5227, can you please also change it?

Comment (fanrenhoo) [reply ddl/ddl_api.go:5236]: sure I can. before did not change it cause new to code need more time more familar, so try change less, and at then not sure if the message same better in the case. I will change L5227 and pull again

Comment (fanrenhoo) [reply ddl/ddl_api.go:5236]: @xiongjiwei I also changed L5227, and use "push -f" forced to commit the new change, please review

Comment (xiongjiwei) [reply ddl/ddl_api.go:5246]: great! thank you for your contribution

BTW, there are some failed tests in CI, you can click the link below to find it, the `check_dev_2` fail seems not related to your changes, just ignore it.

Comment (fanrenhoo) [reply ddl/ddl_api.go:5246]: Thanks! Anything need, just let me know.

Comment (xiongjiwei) [reply ddl/ddl_api.go:5246]: any update for the fail tests? you need to update the tests

Comment (fanrenhoo) [reply ddl/ddl_api.go:5236]: I forced to commit change, can you see the new changes?

Comment (fanrenhoo) [reply ddl/ddl_api.go:5236]: It seem my new change not commited successfully on this branch

Comment (xiongjiwei) [reply ddl/ddl_api.go:5236]: no, I can only see one commit and there are no other changes after I reviewed last time

Comment (fanrenhoo) [reply ddl/ddl_api.go:5247]: @xiongjiwei I finally commit the new change, please review. 
I changed more lines for "_test.go" to make test correct. Actually, its more easy just to change to ErrFileNotFound‘s message same as ErrNoSuchTable, but I thought that will make this message "file not found..." disappear for later usage, so for long run, seem compatible mysql8.0 message and more change is better.

**Review by xiongjiwei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**